server/proxy/expand_test.go

```diff
@@ -0,0 +1,192 @@
+package proxy
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/config"
+	"github.com/mattermost/mattermost-plugin-apps/server/mocks/mock_mmclient"
+)
+
+func TestExpand(t *testing.T) {
+	app := apps.App{
+		BotUserID:   "botid",
+		BotUsername: "botusername",
+		DeployType:  apps.DeployBuiltin,
+		Manifest: apps.Manifest{
+			AppID:       apps.AppID("app1"),
+			DisplayName: "App 1",
+		},
+	}
+
+	userID := "user4567890123456789012345"
+	channelID := "channel7890123456789012345"
+	teamID := "team4567890123456789012345"
+	userChannelRoles := "user_channel_roles"
+	channelMember := &model.ChannelMember{
+		UserId:    userID,
+		ChannelId: channelID,
+		Roles:     userChannelRoles,
+	}
+	teamMember := &model.TeamMember{
+		UserId: userID,
+		TeamId: teamID,
+		Roles:  userChannelRoles,
+	}
+
+	type TC struct {
+		base              apps.Context
+		expectClientCalls func(*mock_mmclient.MockClient)
+		expect            map[apps.ExpandLevel]interface{} // string for err.Error, or apps.ExpandedContext for success
+	}
+
+	for _, field := range []struct {
+		name string
+		tcs  map[string]TC
+	}{
+		{
+			name: "channel_member",
+			tcs: map[string]TC{
+				"happy": {
+					base: apps.Context{
+						ActingUserID:     userID,
+						UserAgentContext: apps.UserAgentContext{ChannelID: channelID},
+					},
+					expectClientCalls: func(client *mock_mmclient.MockClient) {
+						client.EXPECT().GetChannelMember(channelID, userID).Times(4).Return(channelMember, nil)
```

Comment:
Why 4 times? I was expecting 2 times.

Reply:
Good point. Presently expandContext() loads entities for `expand.XXX != ""` and then strips them. This is wrong. I'll file a ticket and cut another PR to fix them all? https://mattermost.atlassian.net/browse/MM-39935